src/TypeScript.UnitTests/Analyzer/EslintBridgeIssueConverterTests.cs

```diff
@@ -73,6 +73,58 @@ public void Convert_IssueConverted()
             convertedIssue.Severity.Should().Be(AnalysisIssueSeverity.Major);
         }
 
+        [TestMethod]
+        [DataRow("aaa", "aaa")]
+        [DataRow("aaa", "AAA")]
+        [DataRow("AAA", "aaa")]
+        [DataRow("AAA", "AAA")]
+        public void Convert_KeyMatchingIsCaseInsensitive(string keyInIssue, string keyInDefinition)
+        {
+            var eslintBridgeIssue = new Issue { RuleId = keyInIssue };
+
+            var ruleDefinitions = new[]
+            {
+                new RuleDefinition
+                {
+                    EslintKey = keyInDefinition,
+                    RuleKey = "sonar rule key"
+                }
+            };
+
+            var testSubject = CreateTestSubject(ruleDefinitions);
+            var convertedIssue = testSubject.Convert("some file", eslintBridgeIssue);
+
+            convertedIssue.RuleKey.Should().Be("sonar rule key");
+        }
+
+        [TestMethod]
+        public void Convert_UnrecognisedESLintKey_Throws()
```

Comment:
Test-as-documentation

This isn't a behaviour we'd thought about explicitly. We could code defensively, but I don't think it's worth it this point.